src/main/java/com/mcsunnyside/coreprotecttnt/Main.java

```diff
@@ -74,6 +75,56 @@ public void run() {
         }.runTaskTimerAsynchronously(this, 0, 20 * 60);
     }
 
+    @EventHandler
+    public void onPlayerInteract(PlayerInteractEvent e) {
+        Block bed = e.getClickedBlock();
+
+        if (bed == null || !e.getAction().equals(Action.RIGHT_CLICK_BLOCK) ||
+                bed.getLocation().getWorld().getEnvironment().equals(World.Environment.NORMAL)) {
+            return;
+        }
+
+        if (!(bed.getType().equals(Material.BLACK_BED) ||
+                bed.getType().equals(Material.BLUE_BED) ||
+                bed.getType().equals(Material.BROWN_BED) ||
+                bed.getType().equals(Material.CYAN_BED) ||
+                bed.getType().equals(Material.GRAY_BED) ||
+                bed.getType().equals(Material.GREEN_BED) ||
+                bed.getType().equals(Material.LIGHT_BLUE_BED) ||
+                bed.getType().equals(Material.LIGHT_GRAY_BED) ||
+                bed.getType().equals(Material.LIME_BED) ||
+                bed.getType().equals(Material.MAGENTA_BED) ||
+                bed.getType().equals(Material.ORANGE_BED) ||
+                bed.getType().equals(Material.PINK_BED) ||
+                bed.getType().equals(Material.PURPLE_BED) ||
+                bed.getType().equals(Material.RED_BED) ||
+                bed.getType().equals(Material.WHITE_BED) ||
+                bed.getType().equals(Material.YELLOW_BED))) {
+            return;
+        }
+
+        Bed data = (Bed) bed.getBlockData();
+        Location location = bed.getLocation();
+        if(data.getPart().equals(Bed.Part.FOOT)) {
+            location = location.add(data.getFacing().getDirection());
+        }
+
+        ignitedBlocks.put(location, e.getPlayer().getName());
```

Comment:
`ignitedBlocks` is meant to contain exploded blocks that are possibly Material. Fire and can cause TNT to explode, i.e. explosion chaining. You should create a separate Map for beds or for all explosive blocks (there is a Respawn Anchor with the same explosion mechanics, and storing both beds and respawn anchors in `ignitedBlocks` is not an option).